cartography/intel/okta/awssaml.py

```diff
@@ -0,0 +1,112 @@
+# Okta intel module - AWS SAML
+import logging
+import re
+
+
+logger = logging.getLogger(__name__)
+
+
+def _parse_regex(regex_string):
+    return regex_string.replace("{{accountid}}", "P<accountid>").replace("{{role}}", "P<role>").strip()
+
+
+def transform_okta_group_to_aws_role(group_id, group_name, mapping_regex):
+    regex = _parse_regex(mapping_regex)
+    matches = re.search(regex, group_name)
+    if matches:
+        accountid = matches.group("accountid")
+        role = matches.group("role")
+        role_arn = f"arn:aws:iam::{accountid}:role/{role}"
+        return {"groupid": group_id, "role": role_arn}
+
+
+def query_for_okta_to_aws_role_mapping(neo4j_session, mapping_regex):
+    """
+    Query the graph for all groups associated with the amazon_aws application and map them to AWSRoles
+    :param neo4j_session: session from the Neo4j server
+    :param mapping_regex: the regex used by the organization to map groups to aws roles
+    """
```

Comment:
Is there a way to replace this with a `merge` statement?  I have some resistance to using `match` to return results from neo4j to use somewhere else; I think there should be a way to avoid the extra back and forth of transferring data.

Reply:
We cannot due that because we need to evaluate a regex on each of the groups and extract values from regex. If the regex was static that might be possible to come up with a query that works, but as people can change the regex we have to run some parsing.